### PR TITLE
Fix issues with `services/enums.py`

### PR DESCRIPTION
**Pull request by BenRKarl**
Add test enforcing `service_client.enums` interface.
Remove non-v2 Enums from `v2/services/enums.py`
Update validation in lazy-loaded enums to check for inner Enum: `Operator`.

Fixes: https://github.com/googleads/google-ads-python/issues/267

### CODE REVIEW

File: google/ads/google_ads/v2/services/enums.py [hunk 1]
```diff
@@ -78,7 +78,10 @@ def __getattribute__(self, name):
                 # If the class attribute has been replaced with something else,
                 # another must have already executed it and created our type.
                 return class_def_src
-            assert class_def_src.startswith('class %s(enum.' % enum_type_name), 'Expected ' + enum_type_name
+            assertion = class_def_src.startswith(
+                'class %s(enum.' % enum_type_name) or class_def_src.startswith(
+                    'class Operator(enum.')
+            assert assertion, 'Expected ' + enum_type_name
```
Comment (BenRKarl) [inline google/ads/google_ads/v2/services/enums.py:84]: `class_def_src` will be the stringified definition of the enum, for example [here](https://github.com/googleads/google-ads-python/blob/2e11de5f0edbb561b6c186aeef95a5605336723a/google/ads/google_ads/v3/services/enums.py#L163). This assertion checks that the string either starts with `'class AccessRole(enum.'` if, for example, the outer class name is `AccessRoleEnum`. This follows the pattern for the majority of our API enums, (see [here](https://github.com/googleapis/googleapis/blob/master/google/ads/googleads/v3/enums/access_role.proto#L31)), where the outer message is `FooBarEnum` and the inner enum is named `FooBar`. In a few cases enums are defined on Operations, for example [`FeedAttributeOperation`](https://github.com/googleapis/googleapis/blob/bf839ae632e0f263a729569e44be4b38b1c85f9c/google/ads/googleads/v1/resources/feed.proto#L175), and the inner enum is called `Operator` so this assertion also allows the patter `'class Operator(enum.'`

File: tests/client_test.py [hunk 2]
```diff
@@ -472,3 +474,60 @@ def test_init_with_logging_config(self):
             Client.GoogleAdsClient(mock_credentials_instance,
                                    self.developer_token, logging_config=config)
             mock_dictConfig.assert_called_once_with(config)
+
+    def test_service_client_dot_enums_property(self):
```
Comment (BenRKarl) [inline tests/client_test.py:478]: This test is a little hard to follow, but basically it tests the `service.enum` property for a service in every version of the API, which implicitly tests the `services/enums.py` file, which exposes the enums used by the property. It loops over each member of `service.enum` and compares it to the raw enum pb2 file found by calling `client.get_type` with the enum's name.

File: google/ads/google_ads/v2/services/enums.py (hunk 1 above)
Comment (wihl) [reply google/ads/google_ads/v2/services/enums.py:84]: This is valuable information so therefore should be in the comments rather than left in this PR.

File: google/ads/google_ads/v2/services/enums.py [hunk 3]
```diff
@@ -5444,19 +5373,19 @@ class ExternalConversionSource(enum.IntEnum):
...
+          UNSPECIFIED (int): Unspecified.
+          UNKNOWN (int): Used for return value only. Represents value unknown in this version.
+          ADD (int): Add the attribute to the existing attributes.
+        """
+        UNSPECIFIED = 0
+        UNKNOWN = 1
+        ADD = 2
 '''
-FeedAttributeOperation = FeedAttributeOperation()
+FeedAttributeOperation = FeedAttributeOperation() # For __getattribute__
```
Comment (wihl) [inline google/ads/google_ads/v2/services/enums.py:5388]: could you either expand or remove this comment?

File: google/ads/google_ads/v3/services/enums.py [hunk 4]
```diff
@@ -78,7 +78,10 @@ def __getattribute__(self, name):
                 # If the class attribute has been replaced with something else,
                 # another must have already executed it and created our type.
                 return class_def_src
-            assert class_def_src.startswith('class %s(enum.' % enum_type_name), 'Expected ' + enum_type_name
+            assertion = class_def_src.startswith(
+                'class %s(enum.' % enum_type_name) or class_def_src.startswith(
+                    'class Operator(enum.')
+            assert assertion, 'Expected ' + enum_type_name
```
Comment (wihl) [inline google/ads/google_ads/v3/services/enums.py:84]: Perhaps expand assertion to include both enum and operator cases?

File: tests/client_test.py (hunk 2 above)
Comment (wihl) [reply tests/client_test.py:478]: Again, this comment belongs in the code rather than in the PR.

File: google/ads/google_ads/v2/services/enums.py (hunk 1 above)
Comment (BenRKarl) [reply google/ads/google_ads/v2/services/enums.py:84]: Sounds good I'll add.

File: tests/client_test.py (hunk 2 above)
Comment (BenRKarl) [reply tests/client_test.py:478]: Ok will add.

File: google/ads/google_ads/v2/services/enums.py (hunk 3 above)
Comment (BenRKarl) [reply google/ads/google_ads/v2/services/enums.py:5388]: This is just for consistency, since it's next to all of the other corresponding lines in this file.

File: google/ads/google_ads/v3/services/enums.py (hunk 4 above)
Comment (BenRKarl) [reply google/ads/google_ads/v3/services/enums.py:84]: Do you mean separate it into two assertions and assert that either one is true?

Comment (wihl) [reply google/ads/google_ads/v3/services/enums.py:94]: No, I mean to expand the assertion message to cover the `operator` case as well. The current message could be ambiguous if I understand correctly.

Comment (BenRKarl) [reply google/ads/google_ads/v3/services/enums.py:94]: Oh I see - I'm not sure that would add much, this is just asserting that the name of the first attribute on the class is of a certain pattern, either `FooBar` for the class `FooBarEnum` or simply `Operator`. In either case it's akin to an `AttributeError`.

**Comment by BenRKarl** [comment]
@wihl updated the backticks to apostrophes
